### PR TITLE
Feature/retrieve rev summary domain

### PR DESCRIPTION
**Pull request by bwruan**

### CODE REVIEW

File: DatamartManagementService/DatamartManagementService.Domain/RevenueSummaryRetrievalService.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    public class RevenueSummaryRetrievalService : IRevenueSummaryRetrievalService
+    {
+        private readonly IRevenueByDateRetrievalRepository _revenueByDateRetrievalRepo;
+
+        public RevenueSummaryRetrievalService(IRevenueByDateRetrievalRepository revenueByDateRetrievalRepo)
+        {
+            _revenueByDateRetrievalRepo = revenueByDateRetrievalRepo;
+        }
+
+        public async Task<List<RofRevenueByDate>> GetRevenueBetweenDates(DateTime startDate, DateTime endDate)
```
Comment (jruan1580) [inline DatamartManagementService/DatamartManagementService.Domain/RevenueSummaryRetrievalService.cs]: Not just RevenueBetweenDates.
After getting all revenue between the dates, i want you to group them, remember?
Display a count of each one and total revenue and gross revenue.

It should be something like `GetRevenueBetweenDatesByPetService`.
Default `string petService = null` as parameter.
If null, then return ALL.
If specified, return specified petService.

Comment (bwruan) [reply DatamartManagementService/DatamartManagementService.Domain/RevenueSummaryRetrievalService.cs]: added

File: DatamartManagementService/DatamartManagementService.Domain/RevenueSummaryRetrievalService.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    public class RevenueSummaryRetrievalService : IRevenueSummaryRetrievalService
+    {
+        private readonly IRevenueByDateRetrievalRepository _revenueByDateRetrievalRepo;
+
+        public RevenueSummaryRetrievalService(IRevenueByDateRetrievalRepository revenueByDateRetrievalRepo)
+        {
+            _revenueByDateRetrievalRepo = revenueByDateRetrievalRepo;
+        }
+
+        public async Task<Dictionary<DateTime, List<RofRevenueByDate>>> GetRevenueBetweenDates(DateTime startDate, DateTime endDate)
```
Comment (jruan1580) [inline DatamartManagementService/DatamartManagementService.Domain/RevenueSummaryRetrievalService.cs]: No not by dates, by PetService.

Comment (bwruan) [reply DatamartManagementService/DatamartManagementService.Domain/RevenueSummaryRetrievalService.cs]: fixed

**Review by jruan1580 — APPROVED** [review]

**Pull request merged**